src/item/FireworkStar.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+/*
+ *
+ *  ____            _        _   __  __ _                  __  __ ____
+ * |  _ \ ___   ___| | _____| |_|  \/  (_)_ __   ___      |  \/  |  _ \
+ * | |_) / _ \ / __| |/ / _ \ __| |\/| | | '_ \ / _ \_____| |\/| | |_) |
+ * |  __/ (_) | (__|   <  __/ |_| |  | | | | | |  __/_____| |  | |  __/
+ * |_|   \___/ \___|_|\_\___|\__|_|  |_|_|_| |_|\___|     |_|  |_|_|
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * @author PocketMine Team
+ * @link http://www.pocketmine.net/
+ *
+ *
+ */
+
+declare(strict_types=1);
+
+namespace pocketmine\item;
+
+use pocketmine\block\utils\DyeColor;
+use pocketmine\color\Color;
+use pocketmine\data\SavedDataLoadingException;
+use pocketmine\nbt\tag\CompoundTag;
+use pocketmine\utils\Binary;
+
+class FireworkStar extends Item{
+
+	protected const TAG_EXPLOSION = "FireworksItem"; //TAG_Compound
+	protected const TAG_CUSTOM_COLOR = "customColor"; //TAG_Int
+
+	protected FireworkRocketExplosion $explosion;
+
+	protected ?Color $customColor = null;
+
+	public function __construct(ItemIdentifier $identifier, string $name){
+		parent::__construct($identifier, $name);
+
+		$this->explosion = new FireworkRocketExplosion(FireworkRocketType::SMALL_BALL(), [DyeColor::BLACK()], [], false, false);
+	}
+
+	public function getExplosion() : FireworkRocketExplosion{
+		return $this->explosion;
+	}
+
+	/** @return $this */
+	public function setExplosion(FireworkRocketExplosion $explosion) : self{
+		$this->explosion = $explosion;
+		return $this;
+	}
+
+	public function getColor() : Color{
```

Comment:
what's the difference between this and the custom colour?

Reply:
By default the firework star gets the color resulting from the mixture of the colors of the explosions it contains, so setting a `customColor` you can choose the color of the item regardless of the color of the explosions.

So, `getColor()` returns the color that is being displayed

This is similar behavior to` Item::getName()` and `Item::getCustomName()`

Reply:
ok, so what's the purpose of this? why can't we just use the colour mix exclusively?

Reply:
Why not use this feature that is enabling mojang? What if users want to use a custom color but don't want to change the explosion?

Reply:
This isn't answering my question. What is the purpose of this? There's no documentation anywhere.